packages/flutter_tools/lib/src/macos/cocoapods.dart

```diff
@@ -349,18 +349,19 @@ class CocoaPods {
   }
 
   void _diagnosePodInstallFailure(ProcessResult result) {
-    if (result.stdout is! String) {
+    final Object? stdout = result.stdout;
+    final Object? stderr = result.stderr;
+    if (stdout is! String || stderr is! String) {
       return;
     }
-    final String stdout = result.stdout as String;
     if (stdout.contains('out-of-date source repos')) {
       _logger.printError(
         "Error: CocoaPods's specs repository is too out-of-date to satisfy dependencies.\n"
         'To update the CocoaPods specs, run:\n'
         '  pod repo update\n',
         emphasis: true,
       );
-    } else if (stdout.contains('ffi_c.bundle') && stdout.contains('LoadError') &&
+    } else if ((stdout.contains('ffi_c.bundle') || stdout.contains('/ffi/')) &&
```

Comment:
@jmagman can you double check this? Your commit had conflicts, but this looked trivial to fix myself.